snakemake.el

```diff
@@ -351,6 +356,46 @@ targets."
              " "))
 
 
+;;; Graphing commands
+
+(defun snakemake-show-dag ()
+  "Create and show dag in emacs-buffer."
+  (interactive)
+  (find-file (snakemake-create-dag)))
+
+(defun snakemake-show-rulegraph ()
+  "Create and show rulegraph in emacs-buffer."
+  (interactive)
+  (find-file (snakemake-create-rulegraph)))
+
+(defun snakemake-create-dag ()
+  "Create dag for workflow.
+
+  Shows all jobs. If the graph becomes too crowded, use the rulegraph function."
+  (interactive)
+  (snakemake--create-dotgraph "dag" "png"))
+
+(defun snakemake-create-rulegraph ()
+  "Create rulegraph for workflow.
+
+  Less crowded than dag option."
+  (interactive)
+  (snakemake--create-dotgraph "dag" "png"))
+
+(defun snakemake--create-dotgraph (graph-type image-type)
+  "Create a graph for the current snakefile."
+  (let ((dot (shell-command-to-string (concat "which " snakemake-graph-program))))
+    (when (equal dot "")
+      (error (concat "You do not have " snakemake-graph-program " on your emacs path!"))))
```

Comment:
I think a "which" dance like this isn't a typical approach.  If
possible, `call-process` (or some other non-shell-command variant)
is used directly and that signals the error if the program isn't
found.  (There is also `executable-find`.)